packages/core/src/types.ts

```diff
@@ -31,14 +31,39 @@ export interface FileProblem {
 //------------------------------------------------------------------------------
 
 /**
- * Represents an AST node or token with location information.
+ * Represents an AST node or token with location information in ESLint format.
  */
-export interface SyntaxElement {
+export interface ESLintSyntaxElement {
 	loc: SourceLocation;
-	range: [number, number];
-	[key: string]: any;
+	range: SourceRange;
+}
+
+/**
+ * Represents an AST node or token with location information containing offsets.
+ */
+export interface LocSyntaxElement {
+	loc: SourceLocationWithOffset;
+}
+
+/**
+ * Represents an AST node or token with location information in unist format.
+ */
+export interface UnistSyntaxElement {
+	position: SourceLocationWithOffset;
+}
+
+/**
+ * Represents an AST node or token with location information in postcss format.
+ */
+export interface PostCSSSyntaxElement {
+	source: SourceLocationWithOffset;
 }
 
+export type SyntaxElement =
+	| ESLintSyntaxElement
+	| UnistSyntaxElement
+	| PostCSSSyntaxElement;
```

Comment:
If we're not going to use anything from the nodes directly, then why restricting them to just these types? I think this should be simply just `Object`.

Reply:
Mentioned in my initial comment:

> Also, I thought that including different formats of SyntaxElement would provide the best developer experience. Otherwise, I'd need to set SourceCode.ast to Object, so there would effectively be no type checking or intellisense available. I'm not sure if that makes the most sense.

Reply:
Okay, playing around with this a bit, it seems like I can still get some good intellisense because `object` can be fulfilled by any other non-primitive type. Just making a few adjustments.